### PR TITLE
feat(github): add tpa sync

### PR DESCRIPTION
**Pull request by elbalexandre**
## Description

- add github tpa sync
- bump required inngest version in test-utils as type where conflicting

## Additional Notes

- [ ] I have test the TPA sync on staging

## Checklist:

Before you create this PR, confirm all the requirements listed below by checking the checkboxes like this (`[x]`).

- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have added tests to cover the new feature or fixes.

### CODE REVIEW

File: apps/github/src/connectors/commons/pagination.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { z } from 'zod';
+
+const nextPageFromLinkSchema = z.preprocess((value) => {
+  if (typeof value !== 'string') return null;
+
+  const nextPageUrl = /<(?<nextPageUrl>[^>]+)>;\s*rel="next"/.exec(value)?.groups?.nextPageUrl;
+  if (!nextPageUrl) return null;
+
+  return new URL(nextPageUrl).searchParams.get('page');
+}, z.coerce.number());
```
Comment (elbalexandre) [inline apps/github/src/connectors/commons/pagination.ts:10]:
```suggestion
}, z.coerce.number().nullable());
```

**Comment by elbalexandre** [comment]
@proquig Uninstallation detection now works, TPA sync is working fine as well.

File: apps/github/.env.test [hunk 2]
```diff
@@ -20,4 +21,9 @@ USERS_SYNC_BATCH_SIZE="5"
 USERS_SYNC_MAX_RETRY="3"
 MAX_CONCURRENT_USERS_SYNC="5"
 
+THIRD_PARTY_APPS_SYNC_CRON="0 0 * * *"
```
Comment (proquig) [inline apps/github/.env.test:24]: You asked me to tell you about setting the timezone, don't forget to add it on vercel

File: apps/github/src/connectors/organization.ts [hunk 3]
```diff
@@ -1,6 +1,7 @@
 import { z } from 'zod';
```
Comment (proquig) [inline apps/github/src/connectors/organization.ts:1]: Skippable. At first I saw the `organization` syntax with a `z` and I thought it was a mistake related to elba organisations. Might be worth to have a github connectors subfolder

File: apps/github/src/inngest/functions/third-party-apps/sync-apps-page.ts [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+      const apps = await Promise.all(
+        result.validInstallations
+          .filter((appInstallation) => appInstallation.suspended_at === null)
+          .map(async (appInstallation) => {
+            const app = await getApp(installationId, appInstallation.app_slug);
+            return formatElbaApp(app, appInstallation, adminIds);
+          })
+      );
+
+      if (result.validInstallations.length) {
```
Comment (proquig) [inline apps/github/src/inngest/functions/third-party-apps/sync-apps-page.ts:102]: Might be worth to log invalid installations to debug them later when needed

File: apps/github/src/inngest/functions/organisations/remove-organisation.ts [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+  async ({ event }) => {
+    const { organisationId } = event.data;
+    const [organisation] = await db
+      .select({
+        region: organisationsTable.region,
+      })
+      .from(organisationsTable)
+      .where(eq(organisationsTable.id, organisationId));
+
+    if (!organisation) {
```
Comment (proquig) [inline apps/github/src/inngest/functions/organisations/remove-organisation.ts:35]: Be careful about this, as your data is not in a step, if somehow the inngest event fails randomly during the connection status, your organisation will be gone and you won't be able to try to set the connection status during the next retries

**Review by proquig — APPROVED** [review]

File: apps/github/src/inngest/functions/organisations/remove-organisation.ts (hunk 5 above)
Comment (elbalexandre) [reply apps/github/src/inngest/functions/organisations/remove-organisation.ts:35]: It's why it's deleted  from db at the end of the function. Am i missing something ?

Comment (proquig) [reply apps/github/src/inngest/functions/organisations/remove-organisation.ts:35]: My bad I misread nevermind 🙈

Comment (proquig) [reply apps/github/src/inngest/functions/organisations/remove-organisation.ts:35]: Now the problem is if elba keeps failing we will never delete the org

**Pull request merged**